builders/ethereum/precompiles/utility/non-specific.md

```diff
@@ -0,0 +1,103 @@
+
+Create a new file called `signMessage.js` in your project directory:
+
+```bash
+touch signMessage.js
+```
+
+Open `signMessage.js` in your code editor and add the following script to initialize Web3 with Moonbase Alpha TestNet, sign and hash the message, and return the signature values:
+
+```js title="signMessage.js"
+--8<-- 'code/builders/ethereum/precompiles/utility/nonspecific/signMessage.js'
+```
+
+Return to your terminal command line to run the script with this command:
+
+```bash
+node signMessage.js
+```
+
+This code will return the following object in the terminal:
+
+--8<-- 'code/builders/ethereum/precompiles/utility/nonspecific/terminal/signature.md'
+
+Save these values as you will need them in the next section.
+
+### Test ECRecoverPublicKey Contract
+
+You can now visit [Remix](https://remix.ethereum.org/){target=\_blank} to test the precompiled contract. Note that you could also use the Web3.js library, but in this case, you can go to Remix to ensure it is using the precompiled contract on the blockchain. The Solidity code you can use to retrieve the public key is the following:
+
+```solidity title="RecoverPublicKey.sol"
+--8<-- 'code/builders/ethereum/precompiles/utility/nonspecific/RecoverPublicKey.sol'
+```
+
+Using the [Remix compiler and deployment](/builders/ethereum/dev-env/remix/){target=\_blank} and with [MetaMask pointing to Moonbase Alpha](/tokens/connect/metamask/){target=\_blank}, you can deploy the contract and call the `recoverPublicKey()` method which returns the public key for the account that signed the message. You can now use this public key value for other cryptographic functions and verifications.
+
+![Returned Public Key on Remix](/images/builders/ethereum/precompiles/utility/nonspecific/nonspecific-1.webp)
+
+## Create a Hash with SHA3FIPS256 {: #create-a-hash-with-sha3fips256 }
+
+SHA3-256 is part of the SHA-3 family of cryptographic hashes codified in [FIPS202](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.202.pdf) that produces an output 256 bits in length. Although the name is similar to SHA256, the SHA-3 family is built with an entirely different algorithm and accordingly produces a different hash output than SHA256 for the same input. You can verify this yourself using this [SHA3-256 Hash Calculator tool](https://md5calc.com/hash/sha3-256){target=\_blank}. After calculating the SHA3-256 output, change the algorithm in the drop-down selector to SHA256 and take note of the resulting output.
+
+Currently there is no SHA3-256 support in Solidity, so it needs to be called with inline assembly. The following sample code can be used to call this precompile.
+
+```solidity
+--8<-- 'code/builders/ethereum/precompiles/utility/eth-mainnet/sha3fips.sol'
```

Comment:
I noticed this uses Solidity version 0.4.21 which dates back to 2016. Unless there is a reason to specify 0.4 can we update this to a more recent one?